acceptable-albatrosses/albatrosses_hub/forums/views.py

```diff
@@ -0,0 +1,338 @@
+from django.shortcuts import render, get_object_or_404
+from django.views.generic import ListView, DetailView, CreateView, UpdateView, DeleteView
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.contrib.auth.models import User
+from django.urls import reverse
+from django.utils import timezone
+from .models import Board, Post, Comment
+
+
+# Create your views here.
+def forumspage(request):
+    """
+    Forums home page, displays all message boards, with information
+    such as the total number of posts in each board.
+    """
+    context = {"boards": Board.objects.all(), "login": request.user.is_authenticated, "page": "forum"}
+    return render(request, "forums/forum.html", context)
+
+
+class PostListView(ListView):
+    """
+    Main page for the message board, displays all posts in the message
+    board.
+    """
+
+    model = Post
+    paginate_by = 5
+    template_name = "forums/board.html"
+    context_object_name = "posts"
+
+    def get_queryset(self):
+        """
+        Queries and return all posts from the same message board.
+        """
+        self.board = get_object_or_404(Board, name=self.kwargs["board"])
+        return Post.objects.filter(board=self.board, comment__isnull=True).order_by("-created_at")
```

Comment:
Your default way of getting a list of comments seems to be in the `"-created_at"` order. It's both used in the `Model` definition as here in the `get_queryset` method you had to define to get this ordering. You could replace that with a `Meta` option in the `Model`, which makes the need for overridden `get_queryset` methods go away (sometimes with a few other tweaks, but there's not here that absolutely requires it).

So, if you add this to the relevant models:

```py
class SomeModel(models.Model):
    # fields here

    class Meta:
        ordering = ["-created_at"]
```

the default ordering of this model would be set to exactly the ordering you want without having to repeat such a `order_by` in custom `filter` methods.